test/dns_dnscmd/dnscmd_test.rb

```diff
@@ -21,25 +22,33 @@ def setup
   end
 
   def test_create_a_record_with_longest_zone_match
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd bar.domain.local host.foo.bar.domain.local. A 192.168.33.33', anything).returns(true)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd', 'bar.domain.local', 'host.foo.bar.domain.local.', 'A', '192.168.33.33', anything).returns(true)
     assert_nil @server.do_create('host.foo.bar.domain.local', '192.168.33.33', 'A')
   end
 
   def test_create_aaaa_record_with_longest_zone_match
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd bar.domain.local host.foo.bar.domain.local. AAAA 2001:db8:deef::1', anything).returns(true)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd', 'bar.domain.local', 'host.foo.bar.domain.local.', 'AAAA', '2001:db8:deef::1', anything).returns(true)
     assert_nil @server.do_create('host.foo.bar.domain.local', '2001:db8:deef::1', 'AAAA')
   end
 
   def test_create_ptr_record
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd 33.168.192.in-addr.arpa 33.33.168.192.in-addr.arpa. PTR host.foo.bar.domain.local.', anything).returns(true)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd', '33.168.192.in-addr.arpa', '33.33.168.192.in-addr.arpa.', 'PTR', 'host.foo.bar.domain.local.', anything).returns(true)
     assert_nil @server.do_create('33.33.168.192.in-addr.arpa', 'host.foo.bar.domain.local', 'PTR')
   end
 
   def test_create_cname_record
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd bar.domain.local alias.foo.bar.domain.local. CNAME host.foo.bar.domain.local', anything).returns(true)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd', 'bar.domain.local', 'alias.foo.bar.domain.local.', 'CNAME', 'host.foo.bar.domain.local', anything).returns(true)
     assert_nil @server.do_create('alias.foo.bar.domain.local', 'host.foo.bar.domain.local', 'CNAME')
   end
 
+  def test_popen_arguments
+    interpreter = Proxy::SETTINGS.x86_64 ? 'c:\windows\sysnative\cmd.exe' : 'c:\windows\system32\cmd.exe'
+    args = ['/RecordAdd', 'bar.domain.local', 'alias.foo.bar.domain.local.', 'CNAME', 'host.foo.bar.domain.local']
+    command = interpreter + ' /c c:\Windows\System32\dnscmd.exe ' + "#{@server}" + args.join(" ")
+    Open3.expects(:popen3).with(command, *args).once.returns(true)
+    Open3.popen3(command, *args)
```

Comment:
Yeah I suggest to create a helper method `popen3` which you can easily mock like:

    @server.expects(:popen3).with(xyz).once.returns([nil, nil, nil])

Then in the main method you also need to add an extra code that is missing so nil exception is not thrown:

```ruby
          response = std_out&.readlines
          response += std_err&.readlines
```

So the method does nothing and returns response which should be `nil`.